feat: added network notice banner as a resuable component

frazarshad opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

<!-- Most PRs should close a specific Issue. All PRs should at least reference one or more Issues. Edit and/or delete the following lines as appropriate (note: you don't need both `refs` and `closes` for the same one): -->

closes: #114 

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

This PR adds `NetworkBanner` (originally implemented in this [PR](https://github.com/Agoric/dapp-psm/pull/94)) as a reusable component for dapps. The purpose of this component is to query the `/network-config` endpoint to find any notices and display them.

The use of this component dapp-econ-gov can be seen in this PR https://github.com/Agoric/dapp-econ-gov/pull/104


**How to test**
- copy `dapp-econ-gov` and `ui-kit` repos and checkout to the `` and `feat/notice-banner` branch respectively
- run `yarn build` in the `ui-kit` repo to create the `dist` directory
- run `yarn add /full/path/to/ui-kit/packages/react-components` in `dapp-econ-gov` to install it as a dependancy.

### Security Considerations

<!-- Does this change introduce new assumptions or dependencies that, if violated, could introduce security vulnerabilities? How does this PR change the boundaries between mutually-suspicious components? What new authorities are introduced by this change, perhaps by new API calls? 
-->

### Scaling Considerations

<!-- Does this change require or encourage significant increase in consumption of CPU cycles, RAM, on-chain storage, message exchanges, or other scarce resources? If so, can that be prevented or mitigated? -->

### Documentation Considerations

<!-- Give our docs folks some hints about what needs to be described to downstream users.

Backwards compatibility: what happens to existing data or deployments when this code is shipped? Do we need to instruct users to do something to upgrade their saved data? If there is no upgrade path possible, how bad will that be for users?

-->

### Testing Considerations

<!-- Every PR should of course come with tests of its own functionality. What additional tests are still needed beyond those unit tests? How does this affect CI, other test automation, or the testnet?
-->

samsiegart commented on packages/react-components/src/lib/components/NoticeBanner.tsx:
```diff
@@ -0,0 +1,90 @@
+  },
+};
+
+const networkConfigAtom = atomWithStorage(
```
I think we should just have the component take the loaded network config JSON as an argument instead of hardcoding and fetching the network configs here. I have some ideas for handling network-configs more holistically in `AgoricProvider`, but I'm still not convinced we want the react components to deal with network configs.

samsiegart reviewed (commented):
Can you add the issue to the PR description please? It would be helpful to see the requirements.

samsiegart commented:
In reference to https://github.com/Agoric/dapp-psm/pull/94, I'm having trouble testing with yarn link as well (probably since we updated to yarn v4). I'm eager to put an example dapp in this package (https://github.com/Agoric/ui-kit/issues/86) so we can test new components easier.

frazarshad commented:
> In reference to [Agoric/dapp-psm#94](https://github.com/Agoric/dapp-psm/pull/94), I'm having trouble testing with yarn link as well (probably since we updated to yarn v4). I'm eager to put an example dapp in this package (#86) so we can test new components easier.

yarn link was also troublesome for me. i've added instructions in the description on how to test these changes

frazarshad commented:
>Can you add the issue to the PR description please? It would be helpful to see the requirements.

No issue exists previously. created one and linked it in the PR

samsiegart commented on packages/react-components/package.json:
```diff
@@ -33,7 +33,8 @@
     "react": "18.2.0",
-    "react-dom": "18.2.0"
+    "react-dom": "18.2.0",
+    "react-icons": "^5.2.1"
```
Can we use something similar from https://heroicons.com/ and copy them into packages/react-components/src/lib/icons? Just trying to avoid adding dependencies and bundle size where possible. Chain-registry and Leap Elements already push the bundle size of this lib over 3MB which really pushes it, I'd like to figure out how to trim this down at some point.

samsiegart commented on packages/react-components/src/lib/components/NoticeBanner.tsx:
```diff
@@ -0,0 +1,41 @@
+export const NoticeBanner = ({
+  notices,
+}: {
+  notices: Array<NetworkNotice>;
```
Let's export the props as "NoticeBannerProps" like https://github.com/Agoric/ui-kit/blob/main/packages/react-components/src/lib/components/AmountInput.tsx#L9. I do this so that they're accessible from the typedocs like https://9418f1f2.ui-kit-dwm.pages.dev/types/_agoric_react_components.AmountInputProps

samsiegart commented on packages/react-components/src/lib/utils/networkConfig.ts:
```diff
@@ -0,0 +1,22 @@
+export type NetworkNotice = {
```
Let's export these from packages/react-components/src/lib/utils/index.ts as well so they're accessible to consumers and in the typedocs

samsiegart reviewed (changes requested):
> No issue exists previously. created one and linked it in the PR

I see, I was wondering to see the motivation for creating this component. I think importing react-components into our inter dapps to use this might be heavy handed for now, but I have no qualms about providing a component like this in general, and it's a good exercise.

> run yarn build in the ui-kit repo to create the dist directory

Heads up, we want to run `yarn prepack` from the top-level of ui-components normally. This will build rpc, web-components, and react-components in order since they depend on eachother, in case you make changes to more than just react-components.


Also, you can remove the "WIP" from the PR now that it's out of draft.

samsiegart reviewed (approved):
Tested on https://github.com/Agoric/dapp-econ-gov/pull/104 and looks great!
